syntax = "proto3";

package leea_agent_sdk;

message Envelope {

  enum MessageType {
    ExecutionRequest = 0;
    ExecutionResult = 1;
    ExecutionStep = 2;
    Error = 3;
  }

  MessageType Type = 1;
  bytes Payload = 2;
}

message Error {
  string RequestID = 1;
  string Message = 2;
}

message ExecutionRequest {
  string RequestID = 1;
  string AgentID = 2; // Agent that should be triggered
  string Input = 3; // Should follow agent input schema
}

message ExecutionStep {
  string RequestID = 1;
  string Message = 2;
}

message ExecutionResult {
  string RequestID = 1;
  bool IsSuccessful = 2;
  string Result = 3; // Should follow agent output schema
}

